src/net/tcp.rs

```diff
@@ -119,7 +121,16 @@ impl TcpStream {
     /// it should already be connected via some other means (be it manually, the
     /// net2 crate, or the standard library).
     pub fn from_stream(stream: net::TcpStream) -> io::Result<TcpStream> {
-        try!(stream.set_nonblocking(true));
+
+        #[cfg(not(target_os = "fuchsia"))]
+        { try!(stream.set_nonblocking(true)); }
+
+        #[cfg(target_os = "fuchsia")]
+        {
+            try!(sys::set_nonblock(
+                ::std::os::unix::io::AsRawFd::as_raw_fd(&stream)));
```

Comment:
I'm not sure how to do this without a cfg in a way that doesn't require adding special `set_nonblock` functions to all the other sys modules. Would you be willing to accept this for the time being, with the understanding that it will be removed when the fix to libstd lands?